### ISSUE TITLE
 Generating files in different directories

### ISSUE THREAD

**Issue by sperochon** [rrousselGit/freezed#553]
**Is your feature request related to a problem? Please describe.**
Hello,

It would be nice to be able to specify .freezed.dart files output directory as done by source_gen ["Generating files in different directories"](https://pub.dev/packages/source_gen#generating-files-in-different-directories) with the _build_extensions_ option.

or (maybe easier/quicker) using a _SharedPartBuilder_ to output to .g.dart files and benefits automatically of _build_extensions_ through _combining_builder_

Current result:
```
lib/models/my_model.dart
lib/models/my_model.g.dart
lib/models/my_model.freezed.dart
```

**Describe the solution you'd like**
```
lib/models/my_model.dart
lib/models/generated/my_model.g.dart
lib/models/generated/my_model.freezed.dart
```

or

```
lib/models/my_model.dart
lib/models/generated/my_model.g.dart   (includes my_model.freezed.dart code)
```

Thanks.

**Comment by rrousselGit** [comment]
I'm not too interested in implementing that. If you need it, I'd be happy to review a PR

**Comment by sperochon** [comment]
I have tried using SharedPartBuilder with combining_builder but it does not work with second pass for serializable annotations inside .freezed.dart files.

So, I've just submitted a PR to source_gen to add the build_extensions option to PartBuilder (https://github.com/dart-lang/source_gen/pull/564).

So waiting for it right now... When accepted, it will then need a light PR to enable it in freezed package.
